x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/termsenum/TermsEnumTests.java

```diff
@@ -30,18 +36,93 @@ public class TermsEnumTests extends ESSingleNodeTestCase {
+    }
+
+    private void checkTermsEnumKeywords(String indexed) throws Exception {
+        String indexName = "test_" + indexed;
+        // only shard is preferred to increase likelihood of having more than one segment, which goes through a different code path
+        int numshards = randomBoolean() ? 1 : randomIntBetween(1, 5);
+        createIndex(indexName, Settings.builder().put("index.merge.enabled", false).put("number_of_shards", numshards).build());
+
+        client().admin()
+            .indices()
+            .preparePutMapping(indexName)
+            .setSource(
+                XContentFactory.jsonBuilder()
+                    .startObject()
+                    .startObject("_doc")
+                    .startObject("properties")
+                    .startObject("keyword")
+                    .field("type", "keyword")
+                    .field("index", indexed)
+                    .endObject()
+                    .endObject()
+                    .endObject()
+                    .endObject()
+            )
+            .get();
+        ensureGreen();
+
+        indexAndRefresh(indexName, "1", "keyword", "Apple");
+        indexAndRefresh(indexName, "2", "keyword", "Able");
+        indexAndRefresh(indexName, "3", "keyword", "Banana");
+        indexAndRefresh(indexName, "4", "keyword", "Blue");
+        indexAndRefresh(indexName, "5", "keyword", "Cheddar");
+        {
+            TermsEnumResponse response = client().execute(TermsEnumAction.INSTANCE, new TermsEnumRequest(indexName).field("keyword")).get();
+            List<String> terms = response.getTerms();
+            assertEquals(5, terms.size());
+            assertThat(terms, contains("Able", "Apple", "Banana", "Blue", "Cheddar"));
+        }
+        {
+            TermsEnumResponse response = client().execute(
+                TermsEnumAction.INSTANCE,
+                new TermsEnumRequest(indexName).field("keyword").string("A")
+            ).get();
+            List<String> terms = response.getTerms();
+            assertEquals(2, terms.size());
+            assertThat(terms, contains("Able", "Apple"));
+        }
+        {
+            TermsEnumResponse response = client().execute(
+                TermsEnumAction.INSTANCE,
+                new TermsEnumRequest(indexName).field("keyword").string("B").searchAfter("Banana")
+            ).get();
+            List<String> terms = response.getTerms();
+            assertEquals(1, terms.size());
+            assertThat(terms, contains("Blue"));
+        }
+    }
+
+    private void indexAndRefresh(String indexName, String id, String field, String value) throws IOException {
+        client().prepareIndex(indexName)
+            .setId(id)
+            .setSource(jsonBuilder().startObject().field(field, value).endObject())
+            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+            .get();
+    }
+
     @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/94378")
```

Comment:
Remove the `@AwaitsFix` annotation?

Reply:
Oh never mind, that's actually for a different bug isn't it

Reply:
Yes, that will be removed in a follow up around terms_enum for the ip type